contracts/modules/src/esdt.rs

```diff
@@ -63,9 +63,9 @@ pub trait EsdtModule {
             ManagedAsyncCallResult::Err(_) => {
                 // return payment to initial caller
                 let initial_caller = self.blockchain().get_owner_address();
-                let egld_returned = self.call_value().egld_value();
-                if *egld_returned > 0u32 {
-                    self.send().direct_egld(&initial_caller, &egld_returned);
+                let egld_returned = self.call_value().egld_value().to_u64().unwrap();
```

Comment:
We shouldn't convert to u64. Might easily not fit. Just dropping the conversion, and it should still work.